app/src/main/java/com/duckduckgo/app/settings/db/SettingsDataStore.kt

```diff
@@ -75,6 +77,13 @@ class SettingsSharedPreferences @Inject constructor(private val context: Context
         get() = preferences.getBoolean(KEY_AUTOCOMPLETE_ENABLED, true)
         set(enabled) = preferences.edit { putBoolean(KEY_AUTOCOMPLETE_ENABLED, enabled) }
 
+    override var appIcon: AppIcon
+        get() {
+            val componentName = preferences.getString(KEY_APP_ICON, AppIcon.DEFAULT.componentName) ?: return AppIcon.DEFAULT
+            return AppIcon.from(componentName)
+        }
+        set(appIcon) = preferences.edit { putString(KEY_APP_ICON, appIcon.componentName) }
```

Comment:
Worth trying `preferences.edit(commit = true)` here since we know the app might crash soon after? Maybe try ensure it has time to flush the changes to disk.

Reply:
👍